op-batcher/batcher/driver.go

```diff
@@ -353,8 +353,15 @@ func (l *BatchSubmitter) publishTxToL1(ctx context.Context, queue *txmgr.Queue[t
 	}
 	l.recordL1Tip(l1tip)
 
+	l1Safe, err := l.safeL1Origin(ctx)
+	if err != nil {
+		l.Log.Error("Failed to query L1 safe origin", "err", err)
+		return err
+	}
+
 	// Collect next transaction data
-	txdata, err := l.state.TxData(l1tip.ID())
+	txdata, err := l.state.TxData(l1tip.ID(), l1Safe)
```

Comment:
I'm wondering about the following simplification. Instead of loading the rollup node's safe L1 origin every time we publish a transaction, we only do it once at startup when we create the `channelManager` and set it as the `l1OriginLastClosedChannel`. (Maybe we also need to do it in a reorg/reset case)

Then, as is already happening, we just update the L1origin in the channelManager every time we close a channel, optimistically assuming that we will publish it.

I think this might simplify the code a fair bit and would alleviate us of the need to thread through the `l1Safe` to `TxData`.